File: module/ownet/python/examples/temperatures.py
#! /usr/bin/env python

import sys
import ownet

if len(sys.argv) != 3:
    print 'temperatures.py server port'
    sys.exit(1)

r = ownet.Sensor('/', server=sys.argv[1], port=int(sys.argv[2]))
e = r.entryList()
s = r.sensorList()
print 'r:', r
print 'r.entryList():', e
print 'r.sensorList():', s

for x in r.sensors():
    if hasattr(x, 'temperature'):
        print x, x.temperature

File: module/ownet/ruby/examples/temperatures.rb
#!/usr/bin/env ruby
$-w = true

require 'ownet'

def usage
    puts 'Usage: temperatures.rb [server [port]]'
end
  
if ARGV.size > 2
  usage
  exit(1)
end

opts = {}
opts[:server] = ARGV[0] if ARGV.size > 0
begin
  opts[:port] = Integer(ARGV[1]) if ARGV.size > 1
rescue ArgumentError
  puts "Port must be an integer"
  usage
  exit(1)
end

root = OWNet::Sensor.new('/', opts)
puts "root: #{root}"

puts "root.entries"
root.each_entry {|e| puts " - #{e}"}

puts "root.sensors"
root.each_sensor {|s| puts " - #{s}"}

root.each_sensor do |s|
  if s.has_attr?('temperature')
    puts "temperature: #{s.temperature}"
    puts "templow: #{s.templow}"
    puts "Setting templow to 25"
    s.templow = 25
    puts "templow: #{s.templow}"
  end
end
        
